raft/raft.go

```diff
@@ -263,7 +263,8 @@ func (r *raft) sendAppend(to uint64) {
 		m.Type = pb.MsgSnap
 		snapshot, err := r.raftLog.snapshot()
 		if err != nil {
-			panic(err) // TODO(bdarnell)
+			r.logger.Debugf("%x failed to send snapshot to %x because snapshot is unavailable (%v)", r.id, to, err)
```

Comment:
can we still check the error before logging. we can still panic on other cases. But it @bdarnell thinks it is OK. I think we can just do the logging. Just try to avoid unnecessary logic changes.